tests/unit/backend/wmg/fixtures/cube.py

```diff
@@ -0,0 +1,154 @@
+    import backend.wmg.data.ontology_labels as ontology_labels
+
+    if ontology_labels.ontology_term_id_labels is None:
+        ontology_labels.__load_ontologies()
+    if ontology_labels.gene_term_id_labels is None:
+        ontology_labels.__load_genes()
+
+    deterministic_term_ids = sorted(ontology_labels.ontology_term_id_labels.keys())
+
+    if dimension_name == "gene_ontology_term_id":
+        return list(sorted(ontology_labels.gene_term_id_labels.keys()))[:dim_size]
+    if dimension_name == "tissue_ontology_term_id":
+        return [term_id for term_id in deterministic_term_ids if term_id.startswith("UBERON")][:dim_size]
+    if dimension_name == "organism_ontology_term_id":
+        return [term_id for term_id in deterministic_term_ids if term_id.startswith("NCBITaxon")][:dim_size]
+    if dimension_name == "cell_type_ontology_term_id":
+        return [term_id for term_id in deterministic_term_ids if term_id.startswith("CL")][:dim_size]
+    if dimension_name == "dataset_id":
+        return [create_dataset(i) for i in range(dim_size)]
+    if dimension_name == "assay_ontology_term_id":
+        return [term_id for term_id in deterministic_term_ids if term_id.startswith("EFO")][:dim_size]
+    if dimension_name == "development_stage_ontology_term_id":
+        return [
+            term_id for term_id in deterministic_term_ids if term_id.startswith("Hsap") or term_id.startswith("MmusDev")
+        ][:dim_size]
+    if dimension_name == "disease_ontology_term_id":
+        return [term_id for term_id in deterministic_term_ids if term_id.startswith("MONDO")][:dim_size]
+    if dimension_name == "ethnicity_ontology_term_id":
+        return [term_id for term_id in deterministic_term_ids if term_id.startswith("HANCESTRO")][:dim_size]
+    if dimension_name == "sex_ontology_term_id":
+        return [term_id for term_id in deterministic_term_ids if term_id.startswith("PATO")][:dim_size]
+    raise AssertionError(f"unknown dimension name {dimension_name}")
+
+
+def create_cube(
+    cube_dir,
+    dim_size=3,
+    attr_vals_fn: Callable[[List[tuple]], List] = random_attr_values,
+    dim_ontology_term_ids_generator_fn: Callable[[str, int], List[str]] = simple_ontology_terms_generator,
+) -> None:
+    if tiledb.array_exists(cube_dir):
+        raise FileExistsError(cube_dir)
+
+    tiledb.Array.create(cube_dir, schema)
+
+    with tiledb.open(cube_dir, mode="w") as cube:
+        n_dims = len(cube_logical_dims)
+        n_coords = dim_size**n_dims
+
+        def dim_domain_values(i_dim: int, dim_size_: int) -> List[str]:
+            dim_name = cube_logical_dims[i_dim]
+            domain_values = dim_ontology_term_ids_generator_fn(dim_name, dim_size_)
+            assert len(set(domain_values)) == dim_size
+            return domain_values
+
+        all_dims_domain_values = [dim_domain_values(i_dim, dim_size) for i_dim in range(n_dims)]
+        # create all possible coordinate values (dim_size ^ n_dims)
+        coords = [
```

Comment:
I would consider breaking this down for readability/clarity
Also why is dim_domain_values nested?

Reply:
add TODO for refactoring

dim_domain_values nesting is just to show that's only be used by the enclosing method